utility/src/main/java/org/oppia/android/util/locale/MachineLocaleImpl.kt

```diff
@@ -50,9 +50,11 @@ class MachineLocaleImpl @Inject constructor(
 
   override fun getCurrentTimeOfDay(): TimeOfDay {
     return when (oppiaClock.getCurrentCalendar().get(Calendar.HOUR_OF_DAY)) {
-      in 4..11 -> TimeOfDay.MORNING
+      in 4..8 -> TimeOfDay.EARLY_MORNING
+      in 9..11 -> TimeOfDay.MORNING
       in 12..16 -> TimeOfDay.AFTERNOON
-      in 0..3, in 17..23 -> TimeOfDay.EVENING
+      in 17..22 -> TimeOfDay.EVENING
+      in 0..3, 23 -> TimeOfDay.NIGHT
```

Comment:
Perhaps 'LATE_NIGHT'?

Reply:
Yes, this is definitely more appropriate.